chain-cli/chaincode-template/e2e/launchpad-api.spec.ts

```diff
@@ -1801,6 +1807,386 @@ describe("LaunchpadContract", () => {
+      expect(response1).toEqual(transactionError());
+      expect(response1.Message).toContain("minFeePortion");
+
+      const rbcConfig2 = new ReverseBondingCurveConfigurationDto();
+      rbcConfig2.minFeePortion = new BigNumber("0.1");
+      rbcConfig2.maxFeePortion = new BigNumber("0.8");
+
+      const invalidSaleDTO2 = new CreateTokenSaleDTO(
+        "HighMaxFee",
+        "HMF",
+        "High Max Fee Test",
+        "www.test.com",
+        new BigNumber("0"),
+        "UnitTest",
+        "none",
+        rbcConfig2
+      );
+      invalidSaleDTO2.websiteUrl = "www.rbctest.com";
+      invalidSaleDTO2.sign(user.privateKey);
+
+      const response2 = await client.Launchpad.CreateSale(invalidSaleDTO2);
+      expect(response2).toEqual(transactionError());
+      expect(response2.Message).toContain("maxFeePortion");
+    });
+
+    test("It should successfully create a sale with valid RBC configuration", async () => {
+      await createGala();
+
+      const rbcConfig = new ReverseBondingCurveConfigurationDto();
+      rbcConfig.minFeePortion = new BigNumber("0.05"); // 5%
+      rbcConfig.maxFeePortion = new BigNumber("0.2"); // 20%
+
+      const saleDTO = new CreateTokenSaleDTO(
+        "ValidRBC",
+        "VRBC",
+        "Valid RBC Token",
+        "www.test.com",
+        new BigNumber("0"),
+        "UnitTest",
+        "Test",
+        rbcConfig
+      );
+      saleDTO.websiteUrl = "www.rbctest.com";
+      saleDTO.sign(user.privateKey);
+
+      const response = await client.Launchpad.CreateSale(saleDTO);
+      expect(response).toEqual(transactionSuccess());
+      expect(response.Data).toBeDefined();
+      expect(response.Data?.vaultAddress).toBeTruthy();
+
+      const saleVaultAddress = response.Data?.vaultAddress;
+      if (!saleVaultAddress) return;
+
+      const fetchSaleDTO = new FetchSaleDto();
+      fetchSaleDTO.vaultAddress = saleVaultAddress;
+
+      const saleDetails = await client.Launchpad.FetchSale(fetchSaleDTO);
+      expect(saleDetails.Data).toBeDefined();
+    });
+
+    test("It should charge the right fee on sell operations", async () => {
+      await createGala();
+
+      const rbcConfig = new ReverseBondingCurveConfigurationDto();
+      rbcConfig.minFeePortion = new BigNumber("0"); // 0%
+      rbcConfig.maxFeePortion = new BigNumber("0.5"); // 50%
+
+      const saleDTO = new CreateTokenSaleDTO(
+        "FeeSellTest",
+        "FST",
+        "Fee Sell Test Token",
+        "www.test.com",
+        new BigNumber("0"),
+        "UnitTest",
+        "Test",
+        rbcConfig
+      );
+
+      saleDTO.websiteUrl = "www.rbctest.com";
+      saleDTO.sign(user.privateKey);
+
+      const saleResponse = await client.Launchpad.CreateSale(saleDTO);
+      expect(saleResponse).toEqual(transactionSuccess());
+      const saleVaultAddress = saleResponse.Data?.vaultAddress;
+      assert(saleVaultAddress, "Sale vault address is undefined");
```

Comment:
`expect`?

Reply:
This assertion is here for type narrowing, otherwise the following line `buyDTO.vaultAddress = saleVaultAddress;` has an error because `Type 'string | undefined' is not assignable to type 'string'`.